spec/integration/plugin/task_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+
+describe 'using the task plugin' do
+  include BoltSpec::Files
+  include BoltSpec::Integration
+
+  def with_boltdir(config: nil, inventory: nil)
+    Dir.mktmpdir do |tmpdir|
+      File.write(File.join(tmpdir, 'bolt.yaml'), config.to_yaml) if config
+      File.write(File.join(tmpdir, 'inventory.yaml'), inventory.to_yaml) if inventory
+      yield tmpdir
+    end
+  end
+
+  let(:config) { { 'modulepath' => ['modules', File.join(__dir__, '../../fixtures/modules')] } }
+
+  let(:plan) do
+    <<~PLAN
+      plan passw() {
+        return(get_targets('node1')[0].password)
+      }
+    PLAN
+  end
+
+  it 'supports a config lookup' do
+    inventory = { 'version' => 2,
+                  'targets' => [
+                    { 'uri' => 'node1',
+                      'config' => {
+                        'ssh' => {
+                          'user' => 'me',
+                          'password' => {
```

Comment:
It might also be worth adding a test that a returned array loads correctly (ie for `run-as-command`). Might also be more trouble than it's worth.